src/arche/rules/coverage.py

```diff
@@ -69,35 +71,17 @@ def get_difference(source_job, target_job):
     return result
 
 
-def compare_scraped_fields(source_df, target_df):
-    source_field_coverage = dict(source_df.count().sort_values(ascending=False))
-    target_field_coverage = dict(target_df.count().sort_values(ascending=False))
-
+def compare_scraped_fields(
+    source_columns: List[str], target_columns: List[str]
+) -> Result:
+    """Find new or missing columns between source_df and target_df"""
     result = Result("Scraped Fields")
-    missing_fields = set(target_df.columns.values) - set(source_df.columns.values)
+    missing_fields = set(target_columns) - set(source_columns)
```

Comment:
if you have arrays this could be much faster for comparison of thousands rows:

import numpy as np
array1 = df1.columns.array
array2 = df2.columns.array
np.setdiff1d(array1, array2)


in this case would be the same

Reply:
This method: setdiff1d will return sorted unique values of comparison 
https://docs.scipy.org/doc/numpy/reference/generated/numpy.setdiff1d.html

Reply:
@manycoding @ivankivanov In fact pandas `Index`es support [set operations](http://pandas.pydata.org/pandas-docs/stable/reference/indexing.html?highlight=index#combining-joining-set-operations) natively. There's no need to convert anything.

![image](https://user-images.githubusercontent.com/3784691/55458350-661c7400-55f5-11e9-8d3e-361716303156.png)